### PR TITLE
Simplify attachment processing in Protocol\ActivityPub\Receiver

### PR DESCRIPTION
**Pull request by MrPetovan**
- Remove indexing attachments by mediatype for later sorting

Follow-up to #10575

### CODE REVIEW

**Comment by annando** [comment]
It looks fine. Have you tried it on your machine?

**Comment by MrPetovan** [comment]
Yes, tried it with the AP conversion tool. It doesn’t change much.

File: src/Protocol/ActivityPub/Receiver.php [hunk 1]
```diff
@@ -1419,18 +1414,7 @@ private static function processObject($object)
 
 		if (in_array($object_data['object_type'], ['as:Audio', 'as:Video'])) {
 			$object_data['alternate-url'] = self::extractAlternateUrl($object['as:url'] ?? []) ?: $object_data['alternate-url'];
-
-			$attachments = self::processAttachmentUrls($object['as:url'] ?? []);
-			foreach ($attachments as $type => $attachment) {
-				$object_data['attachments'][] = [
-					'type' => $type,
-					'mediaType' => $attachment['type'],
```
Comment (annando) [inline src/Protocol/ActivityPub/Receiver.php:1427]: We don't seem to assign the `mediaType` anymore?

File: src/Protocol/ActivityPub/Receiver.php [hunk 2]
```diff
@@ -1290,7 +1290,7 @@ private static function processAttachmentUrls(array $urls): array
 			$filetype = strtolower(substr($mediatype, 0, strpos($mediatype, '/')));
 
 			if ($filetype == 'audio') {
-				$attachments[$filetype] = ['type' => $mediatype, 'url' => $href, 'height' => null, 'size' => null];
+				$attachments[] = ['type' => $mediatype, 'url' => $href, 'height' => null, 'size' => null, 'name' => ''];
```
Comment (annando) [inline src/Protocol/ActivityPub/Receiver.php]: For audio in the previous code the `type` was set to "audio" and the `mediaType` was set to `$mediatype`.

File: src/Protocol/ActivityPub/Receiver.php [hunk 3]
```diff
@@ -1313,14 +1308,14 @@ private static function processAttachmentUrls(array $urls): array
 					continue;
 				}
 
-				$attachments[$mediatype] = ['type' => $mediatype, 'url' => $href, 'height' => $height, 'size' => null];
+				$attachments[$mediatype] = ['type' => $mediatype, 'url' => $href, 'height' => $height, 'size' => null, 'name' => ''];
```
Comment (annando) [inline src/Protocol/ActivityPub/Receiver.php]: The `mediaType` needs to be set.

File: src/Protocol/ActivityPub/Receiver.php [hunk 4]
```diff
@@ -1419,18 +1414,7 @@ private static function processObject($object)
...
-				$object_data['attachments'][] = [
-					'type' => $type,
-					'mediaType' => $attachment['type'],
-					'height' => $attachment['height'],
-					'size' => $attachment['size'],
-					'name' => '',
-					'url' => $attachment['url']
-				];
-			}
+			$object_data['attachments'] = self::processAttachmentUrls($object['as:url'] ?? []);
```
Comment (annando) [inline src/Protocol/ActivityPub/Receiver.php]: We also assign the "attachments" field here: https://github.com/friendica/friendica/blob/develop/src/Protocol/ActivityPub/Receiver.php#L1403

We mustn't replace but add the values here.

**Review by annando — CHANGES_REQUESTED** [review]
Please have a look at the later processing of the attachment variables: https://github.com/friendica/friendica/blob/develop/src/Protocol/ActivityPub/Processor.php#L126-L132

The `mediaType` is a crucial value, so we should set it. The `type` doesn't seem to be used, but I hadn't got a deeper look, so we possibly keep that value at the moment?

File: src/Protocol/ActivityPub/Receiver.php (hunk 1 above)
Comment (MrPetovan) [reply src/Protocol/ActivityPub/Receiver.php:1427]: Indeed, I didn’t check for the exact array key assignments.

**Comment by MrPetovan** [comment]
All feedback has been addressed.

**Comment by annando** [comment]
It looks okay now. Had you been running this code on your system to check if it works well?

**Comment by MrPetovan** [comment]
Here's the final look: https://dev-friendica.mrpetovan.com/display/38a34103-ae53-4e8b-a85a-568ce013e049

**Comment by annando** [comment]
And attachments from Mastodon posts (pictures, videos, ...) are still working?

**Comment by MrPetovan** [comment]
Hmm, I didn't try yet, I don't have a sample for it.

**Comment by annando** [comment]
You could use the branch on your server for half a day or so.

**Comment by MrPetovan** [comment]
Done, let's see how it goes.

**Comment by annando** [comment]
@MrPetovan And? How is it going?

**Comment by MrPetovan** [comment]
Nothing to report concerning attachments. I'm getting pictures and videos as expected.

**Review by annando — APPROVED** [review]

**Pull request merged**